docs/changes/units/15755.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix issue with unicode string representations of units (with superscripts) when printing them in combination with power. The unit now behaves consistently with LaTeX representation and uses the literal short name instead of superscript unicode character in combination with power.
```

Comment:
Since my other comment asks for a change, could you reformat this to a maximum line length of 80? Maybe slightly rephrase too:
```
Fix an issue with unicode string representations of units shown as
superscripts (like degree) when raised to some power. Like for
LaTeX representations, now the superscript unicode character is
replaced by the literal short name before adding the power.
```